<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>666</width>
    <height>557</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Chatter Wave</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
    background-color: #f5f5f5;
}

QGroupBox {
    border: 1px solid #cccccc;
    border-radius: 6px;
    margin-top: 12px;
    font-weight: bold;
    background-color: #ffffff;
}

QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top center;
    padding: 0 5px;
    color: #2c3e50;
}

QPushButton {
    background-color: #3498db;
    color: white;
    border: none;
    padding: 6px 12px;
    border-radius: 4px;
    font-weight: bold;
}

QPushButton:hover {
    background-color: #2980b9;
}

QPushButton:pressed {
    background-color: #1f6dad;
}

QLineEdit {
    border: 1px solid #dddddd;
    border-radius: 4px;
    padding: 5px;
    background-color: #ffffff;
}

QLineEdit:focus {
    border: 1px solid #3498db;
}

QRadioButton {
    spacing: 5px;
}

QRadioButton::indicator {
    width: 15px;
    height: 15px;
}

QTabWidget::pane {
    border: 1px solid #cccccc;
    border-radius: 4px;
    background-color: #ffffff;
}

QTabBar::tab {
    background-color: #e6e6e6;
    border: 1px solid #cccccc;
    border-bottom: none;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
    padding: 6px 10px;
    margin-right: 2px;
}

QTabBar::tab:selected {
    background-color: #ffffff;
    border-bottom: 1px solid #ffffff;
}

QTabBar::tab:hover:!selected {
    background-color: #f0f0f0;
}

QTextEdit {
    border: 1px solid #dddddd;
    border-radius: 4px;
    background-color: #ffffff;
}

QListWidget {
    border: 1px solid #dddddd;
    border-radius: 4px;
    background-color: #ffffff;
}

QListWidget::item {
    padding: 5px;
    border-bottom: 1px solid #f0f0f0;
}

QListWidget::item:selected {
    background-color: #e6f3ff;
    color: #2c3e50;
}

QListWidget::item:hover {
    background-color: #f5f9ff;
}

#publicSendButton {
    background-color: #27ae60;
    min-width: 80px;
}

#publicSendButton:hover {
    background-color: #219955;
}

#connectButton {
    background-color: #9b59b6;
    min-width: 90px;
}

#connectButton:hover {
    background-color: #8e44ad;
}

#publicMessageInput {
    padding: 8px;
    border: 1px solid #dddddd;
    background-color: #ffffff;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="connectionGroupBox">
      <property name="title">
       <string>Connection Settings</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="protocolLayout">
         <item>
          <widget class="QLabel" name="protocolLabel">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Protocol:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="tcpRadio">
           <property name="text">
            <string>TCP</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="udpRadio">
           <property name="text">
            <string>UDP</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="serverLayout">
         <item>
          <widget class="QLabel" name="serverLabel">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Server:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="serverInput">
           <property name="text">
            <string>localhost</string>
           </property>
           <property name="placeholderText">
            <string>Enter server address...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="tcpPortLabel">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>TCP Port:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="tcpPortInput">
           <property name="maximumSize">
            <size>
             <width>60</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>9090</string>
           </property>
           <property name="placeholderText">
            <string>Port</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="udpPortLabel">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>UDP Port:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="udpPortInput">
           <property name="maximumSize">
            <size>
             <width>60</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>9091</string>
           </property>
           <property name="placeholderText">
            <string>Port</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="usernameLabel">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Username:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="usernameInput">
           <property name="text">
            <string>User</string>
           </property>
           <property name="placeholderText">
            <string>Enter your name...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="connectButton">
           <property name="minimumSize">
            <size>
             <width>90</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Connect</string>
           </property>
           <property name="icon">
            <iconset>
             <normaloff>:/icons/connect.png</normaloff>:/icons/connect.png</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSplitter" name="mainSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="leftWidget" native="true">
       <layout class="QVBoxLayout" name="leftLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QGroupBox" name="usersGroupBox">
          <property name="title">
           <string>Online Users</string>
          </property>
          <layout class="QVBoxLayout" name="usersLayout">
           <item>
            <widget class="QLabel" name="doubleClickLabel">
             <property name="text">
              <string>Double-click a user to start a private chat:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QListWidget" name="userList"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QTabWidget" name="chatTabs">
       <property name="tabsClosable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="publicChatTab">
        <attribute name="title">
         <string>Public Chat</string>
        </attribute>
        <layout class="QVBoxLayout" name="publicChatLayout">
         <item>
          <widget class="QTextEdit" name="publicChatDisplay">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="publicMessageLayout">
           <item>
            <widget class="QLineEdit" name="publicMessageInput">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>36</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="placeholderText">
              <string>Type your message to everyone...</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="publicSendButton">
             <property name="minimumSize">
              <size>
               <width>80</width>
               <height>36</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Send</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="protocolInfoGroupBox">
      <property name="title">
       <string>Protocol Information</string>
      </property>
      <layout class="QVBoxLayout" name="protocolInfoLayout">
       <item>
        <widget class="QTextEdit" name="protocolInfo">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>100</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;TCP (Transmission Control Protocol)&lt;/span&gt;&lt;br /&gt;- Connection-oriented protocol that ensures reliable, ordered delivery&lt;br /&gt;- Guarantees message delivery through acknowledgments&lt;br /&gt;- Maintains connection state and handles packet loss automatically&lt;br /&gt;&lt;br /&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;UDP (User Datagram Protocol)&lt;/span&gt;&lt;br /&gt;- Connectionless protocol with no guarantee of delivery&lt;br /&gt;- Faster with less overhead (no handshaking, connection tracking)&lt;br /&gt;- Good for real-time applications where some packet loss is acceptable&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>